pkg/syncer/util.go

```diff
@@ -270,3 +275,19 @@ func initVolumeMigrationService(ctx context.Context, metadataSyncer *metadataSyn
 	}
 	return nil
 }
+
+// isPVStaticallyProvisioned checks if the PersistentVolume has been provisioned statically.
+// Dynamically provisioned PVs have a volume attribute called 'storage.kubernetes.io/csiProvisionerIdentity'
+// in their CSI spec, which is set by external-provisioner.
+// Statically provisioned PVs don't have that attribute.
+func isPVStaticallyProvisioned(pv *v1.PersistentVolume) (bool, error) {
+	if pv == nil {
```

Comment:
This check is not needed since it is already checked here: https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/syncer/metadatasyncer.go#L577